wallet.js

```diff
@@ -0,0 +1,354 @@
+  }
+
+  getTxHistory() {
+    const data = hathorLib.wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in data ? data['historyTransactions'] : {};
+    return historyTransactions;
+  }
+
+  setState(state) {
+    this.state = state;
+    this.emit('state', state);
+  }
+
+  onNewTx(wsData) {
+    const newTx = wsData.history;
+
+    // TODO we also have to update some wallet lib data? Lib should do it by itself
+    const walletData = hathorLib.wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const allTokens = 'allTokens' in walletData ? walletData.allTokens : [];
+
+    let isNewTx = false;
+    if (newTx.tx_id in historyTransactions) {
+      isNewTx = true;
+    }
+
+    hathorLib.wallet.updateHistoryData(historyTransactions, allTokens, [newTx], null, walletData);
+
+    if (isNewTx) {
+      this.emit('new-tx', newTx);
+    } else {
+      this.emit('update-tx', newTx);
+    }
+    return;
+
+    /*
+    // We need to reload it because it was modified by updateHistoryData.
+    const newWalletData = hathorLib.wallet.getWalletData();
+    const { keys } = newWalletData;
+
+    const updatedHistoryMap = {};
+    const updatedBalanceMap = {};
+    const balances = this.getTxBalance(newTx);
+
+    // we now loop through all tokens present in the new tx to get the new history and balance
+    for (const [tokenUid, tokenTxBalance] of Object.entries(balances)) {
+      // we may not have this token yet, so state.tokensHistory[tokenUid] would return undefined
+      const currentHistory = state.tokensHistory[tokenUid] || [];
+      const newTokenHistory = addTxToSortedList(tokenUid, tx, tokenTxBalance, currentHistory);
+      updatedHistoryMap[tokenUid] = newTokenHistory;
+      // totalBalance should not be confused with tokenTxBalance. The latter is the balance of the new
+      // tx, while the former is the total balance of the token, considering all tx history
+      const totalBalance = getBalance(tokenUid);
+      updatedBalanceMap[tokenUid] = totalBalance;
+    }
+    const newTokensHistory = Object.assign({}, state.tokensHistory, updatedHistoryMap);
+    const newTokensBalance = Object.assign({}, state.tokensBalance, updatedBalanceMap);
+
+    hathorLib.storage.setItem('local:tokens:history', newTokensHistory);
+    hathorLib.storage.setItem('local:tokens:balance', newTokensBalance);
+    */
+  };
+
+  /**
+   * Send a transaction with multi tokens
+   * @param {Object} data Array of {'address', 'value', 'token'}
+   */
+  sendMultiTokenTransaction(data) {
+    const txData = {
+      inputs: [],
+      outputs: []
+    }
+
+    const walletData = hathorLib.wallet.getWalletData();
+    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+
+    // First I need an array with all tokens
+    const allTokens = [];
+    for (const d of data) {
+      allTokens.push(d.token);
+    }
```

Comment:
There would be repeated tokens in `allTokens` if different items of `data` have the same token.

Reply:
Fixed. Thanks!